src/Symfony/Component/Security/Http/Authenticator/Passport/Badge/UserBadge.php

```diff
@@ -49,19 +53,26 @@ class UserBadge implements BadgeInterface
      * is thrown). If this is not set, the default user provider will be used with
      * $userIdentifier as username.
      */
-    public function __construct(string $userIdentifier, callable $userLoader = null, array $attributes = null)
+    public function __construct(string $userIdentifier, callable $userLoader = null, array $attributes = null, bool $normalizeIdentifier = null)
```

Comment:
I'd suggest accepting a closure instead, which would be used as the service doing the normalization, allowing to use whatever logic for that + pass it to all places that require the same processing.

Reply:
Indeed. I will update this PR and propose a callable instead.

Reply:
👌🏼 The PR is now up to date and accepts a callable.